api/src/main/java/org/openmrs/api/cache/CacheConfig.java

```diff
@@ -9,35 +9,148 @@
+			// Apply cache type for caches using the 'entity' template 
+			// and add the 'infinispan.cacheContainer.caches' parent.
+			// Skip already defined caches.
+			InputStream fullConfig = buildFullConfig(yaml, configFile,
+				baseConfigBuilder.getNamedConfigurationBuilders().keySet(), cacheType);
+			parser.parse(fullConfig, baseConfigBuilder, null,
+				MediaType.APPLICATION_YAML);
+		}
+		
+		DefaultCacheManager cacheManager = new DefaultCacheManager(baseConfigBuilder, true);
+		return new SpringEmbeddedCacheManager(cacheManager);
+	}
+
+	private static InputStream buildFullConfig(Yaml yaml, URL configFile, Set<String> skipCaches, String cacheType) throws IOException {
+		Map<String, Object> loadedConfig = yaml.load(configFile.openStream());
+
+		Map<String, Object> config = new LinkedHashMap<>();
+		Map<String, Object> cacheContainer = new LinkedHashMap<>();
+		Map<String, Object> caches = new LinkedHashMap<>();
+		Map<String, Object> cacheList = new LinkedHashMap<>();
+		config.put("infinispan", cacheContainer);
+		cacheContainer.put("cacheContainer", caches);
 
-    @Bean(name = "apiCacheManagerFactoryBean")
-    public EhCacheManagerFactoryBean apiCacheManagerFactoryBean(){
-        OpenmrsCacheManagerFactoryBean cacheManagerFactoryBean = new OpenmrsCacheManagerFactoryBean();
-        cacheManagerFactoryBean.setConfigLocation(new ClassPathResource("ehcache-api.xml"));
-        cacheManagerFactoryBean.setShared(false);
-        cacheManagerFactoryBean.setAcceptExisting(true);
+		@SuppressWarnings("unchecked")
+		Map<String, Object> loadedCaches = (Map<String, Object>) loadedConfig.get("caches");
+		for (Map.Entry<String, Object> entry : loadedCaches.entrySet()) {
+			@SuppressWarnings("unchecked")
+			Map<String, Object> value = (Map<String, Object>) entry.getValue();
+			if ("entity".equals(value.get("configuration"))) {
+				Map<Object, Object> cache = new LinkedHashMap<>();
+				cache.put(cacheType, value);
+				if (!skipCaches.contains(entry.getKey())) {
+					cacheList.put(entry.getKey(), cache);
+				}
+			} else {
+				if (!skipCaches.contains(entry.getKey())) {
+					cacheList.put(entry.getKey(), value);
+				}
+			}
+		}
+		if (!cacheList.isEmpty()) {
+			caches.put("caches", cacheList);
+		}
 
-        return cacheManagerFactoryBean;
-    }
+		String configDump = yaml.dump(config);
+		return new ByteArrayInputStream(configDump.getBytes(StandardCharsets.UTF_8));
+	}
 
-    @Bean(name = "apiCacheManager")
-    public CacheManager cacheManager() {
-        return new EhCacheCacheManager(apiCacheManagerFactoryBean().getObject());
-    }
+	public List<URL> getCacheConfigurations() {
+		Resource[] configResources;
+		try {
+			ResourcePatternResolver patternResolver = new PathMatchingResourcePatternResolver();
+			configResources = patternResolver.getResources("classpath*:cache-api.yaml");
+		} catch (IOException e) {
+			throw new IllegalStateException("Unable to find cache configurations", e);
```

Comment:
Does this mean that we throw this exception if we do not find a single cache configuration file?

Reply:
Yes, but there comes one bundled with the api.